rfc/2020/auto-update/supra08.md

```diff
@@ -0,0 +1,40 @@
+- Contribution Name: Auto-Update
+- Implementation Owner: [@supra08](https://github.com/supra08)
+- Start Date: 2020-03-7 (7th Mar, 2020)
+- Target Date: 
+- RFC PR: 
+- Linkerd Issue: [linkerd/linkerd2#1903](https://github.com/linkerd/linkerd2/issues/1903)
+- Reviewers: 
+
+# Summary
+
+[summary]: #summary
+
+The aim of this proposal is to automate the update procedure by implementing a *Kubernetes operator* which will poll a *version-check API* and auto update the control plane and the data plane, also validating its successful completion.
+
+# Problem Statement (Step 1)
+
+[problem-statement]: #problem-statement
+* `Linkerd`, being an active open source project has a high frequency of updates and releases. Maintaining compatible versions in the control and data planes and keeping up with the edge releases is a necessary requirement. Currently the upgrade procedure is primarily manual, with `Linkerd CLI` to be updated and injected with the updated YAML. Maintaining version consistency in the controller pods and data plane proxies requires re-injecting the updated configurations and the developer has to keep up with the edge releases.
```

Comment:
> Maintaining compatible versions in the control and data planes 

It's very rare that there is compatibility issues between the control plane and data plane across versions, as the team works hard to ensure backward compatibliity.

> Maintaining version consistency in the controller pods and data plane proxies requires re-injecting the updated configurations

Users only need to restart their workloads using `kubectl rollout restart`. See https://linkerd.io/2/tasks/upgrade/#upgrade-the-data-plane.

Automated upgrade is the key to remove manual, error prone upgrade steps, while enabling future features like scheduled upgrades, automatic divergence reconciliation etc.

Reply:
Got it!